pkg/operator/start.go

```diff
@@ -0,0 +1,164 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/cilium/ebpf/rlimit"
+	probes_bumblebee_io_v1alpha1 "github.com/solo-io/bumblebee/pkg/api/probes.bumblebee.io/v1alpha1"
+	"github.com/solo-io/bumblebee/pkg/decoder"
+	"github.com/solo-io/bumblebee/pkg/loader"
+	"github.com/solo-io/bumblebee/pkg/operator/internal/cache"
+	"github.com/solo-io/bumblebee/pkg/operator/internal/reconcilers"
+	"github.com/solo-io/bumblebee/pkg/stats"
+	reconcile_v2 "github.com/solo-io/skv2/pkg/reconcile/v2"
+	skv2_stats "github.com/solo-io/skv2/pkg/stats"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	corev1_clients "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const ImageCache = "/tmp/image-cache"
+
+// FIXME:
+// 1. Cosign verification for the images.
+// 2. Authentication for the images.
+// 3. Cache images locally.
+func Start(ctx context.Context) error {
+
+	// Allow the current process to lock memory for eBPF resources.
+	if err := rlimit.RemoveMemlock(); err != nil {
+		return fmt.Errorf("could not raise memory limit (check for sudo or setcap): %v", err)
+	}
+
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return err
+	}
+
+	// Add our probe CRD to the scheme so it is known to the client.
+	if err := probes_bumblebee_io_v1alpha1.AddToScheme(scheme.Scheme); err != nil {
+		return err
+	}
+
+	mgr, err := manager.New(cfg, manager.Options{
+		Scheme:             scheme.Scheme,
+		MetricsBindAddress: "0", // Disable their server as we're using our own.
+	})
+	if err != nil {
+		return err
+	}
+
+	nodeName, ok := os.LookupEnv("NODE_NAME")
+	if !ok {
+		// FIXME: check for pod name, can get node from there.
+		return fmt.Errorf("NODE_NAME environment variable not set, it must be to know where we are running")
+	}
+
+	cli, err := corev1_clients.NewForConfig(cfg)
```

Comment:
nit:
```suggestion
	cli, err := corev1_client.NewForConfig(cfg)
```
clients specifies multiple.